src/containers/CreatePipelineRun/YAMLEditor.js

```diff
@@ -16,29 +16,41 @@ import {
   Button,
   Form,
   FormGroup,
-  InlineNotification
+  InlineNotification,
+  Loading
 } from 'carbon-components-react';
 import yaml from 'js-yaml';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import CodeMirror from '@uiw/react-codemirror';
 import { StreamLanguage } from '@codemirror/language';
 import { yaml as codeMirrorYAML } from '@codemirror/legacy-modes/mode/yaml';
 import { useNavigate } from 'react-router-dom-v5-compat';
 import { createPipelineRunRaw, useSelectedNamespace } from '../../api';
 
-export function CreateYAMLEditor({ code: initialCode = '' }) {
+export function CreateYAMLEditor({
+  code: initialCode,
+  loading = false,
+  loadingMessage = ''
+}) {
   const intl = useIntl();
   const navigate = useNavigate();
   const { selectedNamespace } = useSelectedNamespace();
 
-  const [{ code, isCreating, submitError, validationErrorMessage }, setState] =
+  const [{ isCreating, submitError, validationErrorMessage, code }, setState] =
     useState({
-      code: initialCode,
       isCreating: false,
       submitError: '',
-      validationErrorMessage: ''
+      validationErrorMessage: '',
+      code: initialCode
```

Comment:
nit: please revert this to keep them listed alphabetically, it makes it easier to scan the list